### ISSUE TITLE
"node is null" issue related to font awesome

### ISSUE THREAD

**Issue by bluenote10** [snabbdom/snabbdom#376]
I have tracked down the issue I had (https://github.com/snabbdom/snabbdom/issues/375): Children which are font awesome icons don't seem to have a proper `parentNode` link. This breaks these [two lines](https://github.com/snabbdom/snabbdom/blob/b4d69aea97a7e57e8642d5401f9e36f6e566294f/src/snabbdom.ts#L72-L73), because `parent` will be `null` here.

Minimal code to reproduce (also on [jsfiddle](https://jsfiddle.net/bluenote10/tzztyn25/)):
```javascript
const patch = snabbdom.init([
  snabbdom_class.default,
])
const h = snabbdom.h

vdom1 = h("div", [
  h("div", "Hello World"),
  h("i", {class: {"fas": true, "fa-home": true}})
])

vdom2 =  h("div", [
  h("div", "Hello World"),
])

let oldNode= document.getElementById("root");
setTimeout(() => oldNode = patch(oldNode, vdom1), 1000);
setTimeout(() => oldNode = patch(oldNode, vdom2), 2000);
```

Being a javascript/snabbdom newbie, I don't know if it would make sense to add a null check for `parent` or what would be the proper way to integrate font awesome.

**Comment by caridy** [comment]
We have experienced a similar problem when implementing error boundaries (a la React). Specifically, when recovering from a failure during the rendering process, and trying to recover from it with an alternative view, the vnode tree is messed up (e.g.: some vnodes might or might not be in the DOM), so when attempting to remove them, some of them do not have an actual parentNode (those that failed during the creation hook).

It is slightly different problem, but the exhibit failure. I don't fully understand why is it failing for the `i` with those classes. Can you update your example to not use the minified version of snabbdom, so it is easier to debug?

**Comment by bluenote10** [comment]
Thanks for looking into this!

> Can you update your example to not use the minified version of snabbdom, so it is easier to debug?

I tried to do so (https://jsfiddle.net/tzztyn25/2/), but for some reason the jsfiddle does not play well with the non-minified dist file (it doesn't see the snabbdom functions, and I can't figure out why).

**Comment by blikblum** [comment]
I ve created a codepen at https://codepen.io/blikblum/pen/YLOejE?editors=1011 bug there was no crash

**Comment by bluenote10** [comment]
@blikblum Cool, thanks. There is no crash because it was missing the font awesome js file. If you add `<script src="//use.fontawesome.com/releases/v5.0.7/js/all.js"></script>` you should get the crash as well.

**Comment by caridy** [comment]
is this happening in all browsers? maybe is just a bug outside of snabbdom that we should report.

**Comment by bluenote10** [comment]
I tried Firefox and Chrome and they both have the issue. Could it be something like that:

- Snabbdom internally keeps a reference to the `<i>` element.
- The font awesome js code replaces the `<i>` element by the corresponding `<svg>` tag.
- Now the `<i>` element is no longer part of the DOM, so when snabbdom tries to access the parent of `<i>` there is none.

I also tested if this is a general issue with `svg` tags, but static `svg` element seem to work fine.
